shop/src/main/java/ro/msg/learning/shop/strategy/SingleLocation.java

```diff
@@ -0,0 +1,51 @@
+package ro.msg.learning.shop.strategy;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import ro.msg.learning.shop.dto.ProductOrderDTO;
+
+import ro.msg.learning.shop.model.Stock;
+import ro.msg.learning.shop.repository.IStockRepository;
+
+import javax.transaction.Transactional;
+import java.util.*;
+
+
+
+@RequiredArgsConstructor
+@Transactional
+public class SingleLocation implements ILocationStrategy{
+    @Autowired
```

Comment:
If you have the `@RequiredArgsConstructor` you don't need to use `@Autowired` anymore. You should remove it and put `private final` before the `stockRepository`